### PR TITLE
Fix API links for Scala.js 1.0.0

### PR DESCRIPTION
**Pull request by MaximeKjaer**
I noticed that a few API links were broken, so here's a PR to fix them.

Some parts of the documentation might need to be rewritten for 1.0.0. I only fixed the links I could fix; if I didn't find something in the 1.0.0 API docs, I didn't fix the link.

### CODE REVIEW

File: doc/project/js-environments.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ jsEnv := PhantomJSEnv(args = Seq("arg1", "arg2")).value
 {% endhighlight %}
 
 For more options of the PhantomJS environment, see
-[the Scaladoc of `PhantomJSEnv`]({{ site.production_url }}/api/sbt-scalajs-env-phantomjs/1.0.0/org/scalajs/jsenv/phantomjs/sbtplugin/PhantomJSEnvPlugin$$autoImport$.html).
+[the Scaladoc of `PhantomJSEnv`]({{ site.production_url }}/api/sbt-scalajs-env-phantomjs/{{ site.versions.scalaJS }}/org/scalajs/jsenv/phantomjs/sbtplugin/PhantomJSEnvPlugin$$autoImport$.html).
```
Comment (sjrd) [inline doc/project/js-environments.md]: This should remain hard-coded 1.0.0. The version of sbt-scalajs-env-phantomjs is independent of the version of Scala.js.

File: doc/interoperability/types.md [hunk 2]
```diff
@@ -283,7 +283,7 @@ want to. The basic entry point is `js.Dynamic.global`, which is a dynamically
 typed view of the JavaScript global scope. You can select any global variable
 of JavaScript as a a member of `js.Dynamic.global`, e.g.,
 `js.Dynamic.global.Math`, which will be typed as a
-[`js.Dynamic`]({{ site.production_url }}/api/scalajs-library/latest/#scala.scalajs.js.Dynamic).
+[`js.Dynamic`]({{ site.production_url }}/api/scalajs-library/{{ site.versions.scalaJS }}//scala/scalajs/js/Dynamic.html).
```
Comment (sjrd) [inline doc/interoperability/types.md]:
```suggestion
[`js.Dynamic`]({{ site.production_url }}/api/scalajs-library/{{ site.versions.scalaJS }}/scala/scalajs/js/Dynamic.html).
```

File: doc/interoperability/sjs-defined-js-classes.md [hunk 3]
```diff
@@ -445,7 +445,7 @@ With run-time overload resolution, however, the type tests are executed again, a
 
 ### `js.constructorOf[C]`
 
-To obtain the JavaScript constructor function of a JS class (native or not) without instantiating it nor exporting it, you can use [`js.constructorOf[C]`]({{ site.production_url }}/api/scalajs-library/latest/#scala.scalajs.js.package@constructorOf[T<:scala.scalajs.js.Any]:scala.scalajs.js.Dynamic), whose signature is:
+To obtain the JavaScript constructor function of a JS class (native or not) without instantiating it nor exporting it, you can use [`js.constructorOf[C]`]({{ site.production_url }}/api/scalajs-library/{{ site.versions.scalaJS }}/#scala.scalajs.js.package@constructorOf[T<:scala.scalajs.js.Any]:scala.scalajs.js.Dynamic), whose signature is:
```
Comment (sjrd) [inline doc/interoperability/sjs-defined-js-classes.md]:
```suggestion
To obtain the JavaScript constructor function of a JS class (native or not) without instantiating it nor exporting it, you can use [`js.constructorOf[C]`]({{ site.production_url }}/api/scalajs-library/{{ site.versions.scalaJS }}/scala/scalajs/js/index.html#constructorOf[T<:scala.scalajs.js.Any]:scala.scalajs.js.Dynamic), whose signature is:
```

Comment (MaximeKjaer) [reply doc/interoperability/sjs-defined-js-classes.md]: Fixed in 1349916893f6b79555756ee40b5fc33ff9b99d44

File: doc/project/js-environments.md (hunk 1 above)
Comment (MaximeKjaer) [reply doc/project/js-environments.md]: All right, thought it was a typo! Fixed in 1349916893f6b79555756ee40b5fc33ff9b99d44

**Comment by MaximeKjaer** [comment]
Thanks for the review! I'll let you do a squash when merging.

**Pull request merged**